docs/setup/gcp-pub-sub.md

```diff
@@ -1,63 +1,95 @@
-                  "id": "unique_event_id",
-                  "time": "event_time",
-                  "datacontenttype": "type_of_data",
-                  "subject": "name_of_the_configuration_within_event_source"
-                },
-                "data": {
-                    "id": "message id",
-                    // Attributes represents the key-value pairs the current message
-                    // is labelled with.
-                    "attributes": "key-values",
-                    "publishTime": "// The time at which the message was published",
-                  	"body": "body refers to the message data",
-                }
-            }
+```js
+{
+    "context": {
+      "type": "type_of_event_source",
+      "specversion": "cloud_events_version",
+      "source": "name_of_the_event_source",
+      "id": "unique_event_id",
+      "time": "event_time",
+      "datacontenttype": "type_of_data",
+      "subject": "name_of_the_configuration_within_event_source"
+    },
+    "data": {
+        "id": "message id",
+        // Attributes represents the key-value pairs the current message
+        // is labelled with.
+        "attributes": "key-values",
+        "publishTime": "// The time at which the message was published",
+        "body": "body refers to the message data",
+    }
+}
+```
 
 ## Specification
 
-GCP PubSub event-source specification is available [here](https://github.com/argoproj/argo-events/blob/master/api/event-source.md#pubsubeventsource).
+GCP Pub/Sub event-source specification is available [here](https://github.com/argoproj/argo-events/blob/stable/api/event-source.md#pubsubeventsource).
 
 ## Setup
 
 1. Fetch the project credentials JSON file from GCP console.
 
+     If you use Workload Identity, you can skip this and next steps.
+
 1. Create a K8s secret called `gcp-credentials` to store the credentials file
 
-        apiVersion: v1
-        data:
-          key.json: <YOUR_CREDENTIALS_STRING_FROM_JSON_FILE>
-        kind: Secret
-        metadata:
-          name: gcp-credentials
-          namespace: argo-events
-        type: Opaque
+    ```yaml
+    apiVersion: v1
+    data:
+      key.json: <YOUR_CREDENTIALS_STRING_FROM_JSON_FILE>
+    kind: Secret
+    metadata:
+      name: gcp-credentials
+      namespace: argo-events
+    type: Opaque
+    ```
 
 1. Create the event source by running the following command.
-   
-           kubectl apply -n argo-events -f https://raw.githubusercontent.com/argoproj/argo-events/stable/examples/event-sources/gcp-pubsub.yaml
+
+    ```sh
+    kubectl apply -n argo-events -f https://raw.githubusercontent.com/argoproj/argo-events/stable/examples/event-sources/gcp-pubsub.yaml
+    ```
+
+    If you use Workload Identity, omit `credentialSecret` field. Instead don't forget to configure appropriate service account (see [example](https://github.com/argoproj/argo-events/blob/stable/examples/event-sources/gcp-pubsub.yaml)).
```

Comment:
Workload Identity works by connecting Google Service Account (GSA) and Kubernetes Service Account (KSA) so that KSA can obtain GSA's token.

This connection is done by putting an annotation to KSA like below:
(NB: in addition to this, appropriate IAM should be set on GCP side, but no extra RoleBinding is needed on k8s side)

```yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  namespace: argo-events
  name: tmshn-tester
  annotations:
    iam.gke.io/gcp-service-account: argo-events-tmshn-tester@my-gcp-project.iam.gserviceaccount.com
```

ref: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity

This setup is very basic on using Workload Identity, but we need to know which KSA runs the event source pod.
So at least I want to document that even if we don't encourage users to create new KSA.

Is that `default` or `argo-events-sa` ?

Reply:
You are right, at my first sight, I thought the doc was something about "configure proper service account to access the secret...", please ignore that comment. However, I think the link should be pointing to https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity.